cms/server/AdminWebServer.py

```diff
@@ -759,17 +1087,20 @@ def post(self, task_id):
         if try_commit(self.sql_session, self):
             self.redirect("/task/%s" % task_id)
         else:
-            self.redirect("/add_statement/%s" % task_id)
+            self.redirect(fallback_page)
 
 
 class DeleteStatementHandler(BaseHandler):
     """Delete a statement.
 
     """
-    def get(self, statement_id):
+    def get(self, task_id, statement_id):
         statement = self.safe_get_item(Statement, statement_id)
-        task = statement.task
-        self.contest = task.contest
+        task = self.safe_get_item(Task, task_id)
+
+        # Additional check.
+        if task is not statement.task:
+            raise tornado.web.HTTPError(404)
```

Comment:
Ah, right, I mistakenly thought this check was for possible database corruption. Yes, the 404 is appropriate, the comment is... perfectible :) Can you change it everywhere to  something like "Protect against URLs providing incompatible parameters"?